app/api/business/questions_business.py

```diff
@@ -0,0 +1,130 @@
+
+    if answers:
+        answersCount = len(answers)
+    else:
+        answersCount = len(brief_clarification_question_service.get_answers(brief_id))
+
+    return {
+        "questions": questionsCount,
+        "answers": answersCount
+    }
+
+
+def get_question(brief_id, question_id):
+    brief = briefs.find(id=brief_id).one_or_none()
+    if not brief:
+        raise NotFoundError("Invalid brief id '{}'".format(brief_id))
+
+    question = brief_question_service.find(id=question_id).one_or_none()
+    question_result = None
+    if question:
+        question_result = {
+            "id": question.id,
+            "data": question.data
+        }
+
+    return {
+        "question": question_result,
+        "brief": {
+            "title": brief.data.get('title'),
+            "id": brief.id,
+            "questionsCloseAt": brief.questions_closed_at,
+            "internalReference": brief.data.get('internalReference')
+        }
+    }
+
+
+def get_questions(brief_id):
+    brief = briefs.find(id=brief_id).one_or_none()
+    if not brief:
+        raise NotFoundError("Invalid brief id '{}'".format(brief_id))
+
+    questions = brief_question_service.get_questions(brief_id)
+
+    return {
+        "questions": questions,
+        "brief": {
+            "title": brief.data.get('title'),
+            "id": brief.id,
+            "internalReference": brief.data.get('internalReference')
+        },
+        "questionCount": get_counts(brief_id, questions=questions)
+    }
+
+
+def get_answers(brief_id):
+    brief = briefs.find(id=brief_id).one_or_none()
+    if not brief:
+        raise NotFoundError("Invalid brief id '{}'".format(brief_id))
+
+    answers = brief_clarification_question_service.get_answers(brief_id)
+
+    return {
+        "answers": answers,
+        "brief": {
+            "title": brief.data.get('title'),
+            "id": brief.id,
+            "internalReference": brief.data.get('internalReference')
+        },
+        "questionCount": get_counts(brief_id, answers=answers)
+    }
+
+
+def publish_answer(user_info, brief_id, data):
+    brief = briefs.get(brief_id)
+    if not brief:
+        raise NotFoundError("Invalid brief id '{}'".format(brief_id))
+
+    if not briefs.has_permission_to_brief(user_info.get('user_id'), brief.id):
+        raise UnauthorisedError('Unauthorised to publish answer')
+
+    publish_question = data.get('question')
+    if not publish_question:
+        raise ValidationError('Question is required')
+
+    answer = data.get('answer')
+    if not answer:
+        raise ValidationError('Answer is required')
+
+    brief_clarification_question = brief_clarification_question_service.save(BriefClarificationQuestion(
+        _brief_id=brief_id,
+        question=publish_question,
+        answer=answer,
+        user_id=user_info.get('user_id')
+    ))
+
+    question_id = data.get('questionId')
+    if question_id:
+        question = brief_question_service.get(question_id)
```

Comment:
This is not overly damaging, but it is possible to change this `questionId` value in the request to any other valid id value and have the system flag that question as answered, i.e. there is no validation the id is owned by the user in question. Maybe a check that the brief related to this question id is associated with the user?

Reply:
check that the question being answered has the same brief id.